src/Engine/Objects/PlayerEnums.h

```diff
@@ -451,6 +439,27 @@ inline PLAYER_CLASS_TYPE getTier3DarkClass(PLAYER_CLASS_TYPE classType) {
     return (PLAYER_CLASS_TYPE)(classType - tier + 3);
 }
 
+/**
+ * Get priomotions of higher tier class relative to given one.
+ *
+ * Base class is of tier 1.
+ * After initial promotion class becomes tier 2.
+ * Tier 2 class is promoted through light or dark path to tier 3 class.
+ *
+ * @param classType     Character class.
+ */
+inline Segment<PLAYER_CLASS_TYPE> getClassPromotions(PLAYER_CLASS_TYPE classType) {
+    int tier = classType % 4;
+
+    if (tier == 1) {
```

Comment:
I don't think this is correct.

E.g. with `PLAYER_CLASS_KNIGHT` you get tier == 0 and return an empty `Segment`.